### PR TITLE
React redux branch

### PR DESCRIPTION
**Pull request by soesandarwin2201**
## What does this PR do?
This PR is raised to use Redux store to display books, add some UI elements to components and make them dispatch actions.
General requirements 

## Description of the task to be completed

- [x] Import Redux Provider and  store in the main component of  React app.
- [x] Connect app to the Redux store.
- [x] Use useSelector to consume the state.
- [x] In React component responsible for adding new books, add functionality for "Add new" button click:
- [x] Import Redux useDispatch.
- [x] Set the data inputs in the local React state 
- [x] Takes the inputs from the state and generate a unique id and add them to an object.
- [x] In r React component responsible for removing books, add a functionality for "Remove" button click:
- [x] Import Redux useDispatch.
- [x] Dispatch a corresponding action
- [x] Import Redux useDispatch.
- [x] Dispatch a corresponding action 
- [x] fix the linters errors

## How should it be tested
Clone the repository
```
git clone  https://github.com/soesandarwin2201/Book_Store_React_Project
```
Navigate to the folder
```
cd Book_Store_React_Project
```
Checkout the branch
```
git checkout react-redux-branch
```
#### Thanks for reviewing my PR :blush:

### CODE REVIEW

File: src/components/book.jsx [hunk 1]
```diff
@@ -1,18 +1,31 @@
...
+function Book({ book }) {
+  const dispatch = useDispatch();
   return (
-    <>
     <section className="container">
-     <h2>{title}</h2>
-     <p>{author}</p>
+      <h2>{book.title}</h2>
+      <p>{book.author}</p>
+      <Categories />
```
Comment (julie-ify) [inline src/components/book.jsx]: - [x] Kindly remove this Categories component from here. It is displayed and repeated on the Books homepage.

  ifdo

- [ ] When I click the "categories" on the navbar, nothing happens. I should be navigated to the categories component. In your **App.js** component kindly define your Categories route accordingly and ensure that when the "check status" button is clicked the text is displayed.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @soesandarwin2201,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/book.jsx (hunk 1 above)
Comment (soesandarwin2201) [reply src/components/book.jsx]: thanks for your time and your effort, and thanks for pointing out the fact in details. I will improve my code with your suggestions.

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


## Optional suggestions


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**